app/src/main/java/org/fossasia/susi/ai/chat/ParseSusiResponseHelper.kt

```diff
@@ -42,9 +42,23 @@ class ParseSusiResponseHelper {
 
             Constant.ANSWER -> try {
                 answer = susiResponse.answers[0].actions[i].expression
-                val urlList = extractUrls(answer)
-                isHavingLink = true
-                if (urlList.isEmpty()) isHavingLink = false
+
+                //get the Urls stored in 'data' of the answer object
+                val text = susiResponse.answers[0].data[0].get("object") //requires api warning suppressed
+                if (text != null) {
+                    val urlList = extractUrls(text)
+
+                    //appending the Urls to the answer
+                    //numLinks is the maximum number of links to be appended
+                    val numLinks = 1
+                    for (link in urlList.indices) {
+                        if (link < numLinks)
+                            answer += "\n" + urlList[link]
+                    }
```

Comment:
This is not C, directly iterate through link

Also, you don't even need to do that

```kotlin
isHavingLink = true
if (urlList.isNotEmpty()) {
    answer += "\n" + urlList[0]
} else {
    isHavingLink = false
}
```

Reply:
This was done because I didn't know how many links should preferably to be included in the answer. Doing it this way leaves an option to easily change the number if required

Reply:
What'll happen if we show all links?

Reply:
For simple queries, there are a lot of links. Showing all the links will make the answer look messy and the user will have to scroll a lot to find the appropriate result.
Also the preview image for the link is also not displayed which makes the result even more ugly

Reply:
OK, then we'll go with only 1 link

```kotlin
isHavingLink = true
if (urlList.isNotEmpty()) {
    answer += "\n" + urlList[0]
} else {
    isHavingLink = false
}
```